### PR TITLE
Support multitenancy for `Stand`

### PR DESCRIPTION
**Pull request by armiol**
This PR introduces a multitenancy support in `Stand`.

**BREAKING CHANGES**

* `ActorRequestFactory` has been introduced as a common ground for instantiation of `Command`, `Query` and `Topic` instances. `CommandFactory` has been removed.
* `Stand` public API has been updated to work with `Topic` instead of working directly with `Target`s, as previously.
* In order to post updates to a `Stand` instance, `StandFunnel` now requires `CommandContext` of the `Command`, which triggered the entity update.
* `Stand.update()` now takes `EntityStateEnvelope` instead of `id`, `version` and `state` args, as previously.

**Other changes**

* `Delivery` handling has been also adjusted to take potential multitenancy of the `BoundedContext` instance.
* Processing of command output objects is now done in scope of `TenantAwareOperation` (see `FailureSubscriber` and `EventSubscriber`).

**Request creation**
```
final ActorRequestFactory factory = ActorRequestFactory.newBuilder()
                                                       .setActor(myUserId)
                                                       .setTenantId(myTenantId)  // optional; not set in a single-tenant apps.
                                                       .setTimezone(timezone)   // UTC by default.
                                                       .build();
// Commands:
final EditProject cmdMessage = ...;
final Command command = factory.command().create(cmdMessage);

// Queries:
final Query allProjects = factory.query().all(Project.class);

//Topics:
final Topic updatesOfMyProject = factory.topic().someOf(Project.class, projectId);
```

**Subscribing to changes**

```
final Stand stand = ...;
final Subscription subscription = stand.subscribe(updatesOfMyProject);

// activate the subscription to start receiving the updates via the callback:
stand.activate(subscription, new Stand.EntityUpdateCallback() { ... });

// cancel the subscription to stop receiving the updates.
stand.cancel(subscription);
```

**NEXT**

After this PR the following changes are planned.

* Review `CommandContext` to incorporate `ActorContext`.
* Validate input parameters of `Stand` public API to ensure they are valid `Message`s according to their Protobuf definitions.
* Generate identifiers for `Topic` and `Query` in `ActorRequestFactory`, so that they are more human-readable.

### CODE REVIEW

**Comment by armiol** [comment]
@alexander-yevsyukov PTAL.

File: server/src/main/java/org/spine3/server/failure/FailureSubscriber.java [hunk 1]
```diff
@@ -38,15 +40,28 @@
...
      * Cached set of the failure classes this subscriber is subscribed to.
      */
     @Nullable
     private Set<FailureClass> failureClasses;
 
     @Override
-    public void dispatch(FailureEnvelope envelope) {
-        handle(envelope.getMessage(), envelope.getCommandMessage(), envelope.getCommandContext());
+    public void dispatch(final FailureEnvelope envelope) {
+        final Command originCommand = envelope.getOuterObject()
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/failure/FailureSubscriber.java]: Please fix wrapping here.

File: server/src/main/java/org/spine3/server/stand/Stand.java [hunk 2]
```diff
@@ -142,62 +157,103 @@ public static Builder newBuilder() {
...
+
+                if (isAggregateUpdate) {
+                    final Optional<Version> entityVersion = envelope.getEntityVersion();
+                    checkState(entityVersion.isPresent(),
+                               "The aggregate version must be set in order to update Stand. " +
+                                       "Actual envelope: {}", envelope);
+
+                    @SuppressWarnings("OptionalGetWithoutIsPresent")    // checked above.
+                    final Version versionValue = entityVersion.get();
+
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/stand/Stand.java]: Please remove excessive empty line.

File: server/src/test/java/org/spine3/server/command/EventFactoryShould.java [hunk 3]
```diff
@@ -33,11 +33,12 @@
...
 
-    private final CommandFactory commandFactory = TestCommandFactory.newInstance(getClass());
+    private final ActorRequestFactory requestFactory =
+            TestActorRequestFactory.newInstance(getClass());
 
     @Test(expected = NullPointerException.class)
     public void require_producer_id_in_builder() {
-        final CommandContext ctx = commandFactory.createCommand(Empty.getDefaultInstance())
+        final CommandContext ctx = requestFactory.command().create(Empty.getDefaultInstance())
                                                  .getContext();
```
Comment (alexander-yevsyukov) [inline server/src/test/java/org/spine3/server/command/EventFactoryShould.java]: Please fix wrapping here.

**Comment by alexander-yevsyukov** [comment]
@armiol, have a look at `TopicFactory`. It needs Javadoc for the class.

**Comment by armiol** [comment]
@alexander-yevsyukov thanks for the review. The issues are now addressed.

Regarding `TopicFactory` — I'm not sure I get it, as this class has been deleted at all. I have looked through the newly introduced classes and found no spots, where the docs were missing. So let's discuss.

**Pull request merged**